python/cudf/cudf/core/tools/numeric.py

```diff
@@ -0,0 +1,241 @@
+    elif is_list_dtype(dtype) or is_struct_dtype(dtype):
+        raise ValueError("Input does not support nested datatypes")
+    elif is_numerical_dtype(dtype):
+        pass
+    else:
+        raise ValueError("Unrecognized datatype")
+
+    if downcast:
+        downcast_type_map = {
+            "integer": list(np.typecodes["Integer"]),
+            "signed": list(np.typecodes["Integer"]),
+            "unsigned": list(np.typecodes["UnsignedInteger"]),
+        }
+        float_types = list(np.typecodes["Float"])
+        idx = float_types.index(np.dtype(np.float32).char)
+        downcast_type_map["float"] = float_types[idx:]
+
+        type_set = downcast_type_map[downcast]
+        is_similar_kind_cast = (
+            col.dtype.kind == "f" and downcast == "float"
+        ) or (
+            col.dtype.kind in {"i", "u"}
+            and downcast in {"integer", "signed", "unsigned"}
+        )
+
+        for t in type_set:
+            downcast_dtype = np.dtype(t)
+            if downcast_dtype.itemsize <= col.dtype.itemsize:
+                if is_similar_kind_cast:
+                    if col.can_cast_safely(downcast_dtype):
+                        col = libcudf.unary.cast(col, downcast_dtype)
+                        break
+                else:
+                    if _can_cast_no_overflow(col, downcast_dtype):
+                        # Round off for float -> (u)int
+                        if col.dtype.kind == "f" and downcast_dtype.kind in {
+                            "i",
+                            "u",
+                        }:
+                            # TODO: use libcudf round once
+                            # https://github.com/rapidsai/cudf/pull/6562 merges
+                            col = col.round()
+                        col = libcudf.unary.cast(col, downcast_dtype)
+                        break
+
+    if isinstance(arg, (cudf.Series, pd.Series)):
+        return cudf.Series(col)
+    else:
+        return col.to_array(fillna="pandas")
+
+
+def _convert_str_col(col, errors):
+    """
+    Converts a string column to numeric column
+
+    Converts to integer column if all strings are integer-like (isinteger.all)
+    Otherwise, converts to float column if all strings are float-like (
+        isfloat.all)
+    If error == 'coerce', fill non-numerics strings with null
+
+    Parameters
+    ----------
+    col : The string column to convert, must be string dtype
+    errors : {'raise', 'ignore', 'coerce'}, same as to_numeric
+
+    Returns
+    -------
+    Converted numeric column
+    """
+    if not is_string_dtype(col):
+        raise TypeError("col must be string dtype.")
+
+    is_integer = col.str().isinteger()
+    if is_integer.all():
+        return col.as_numerical_column(dtype=np.dtype("i8"))
+
+    # Account for `inf` strings
+    col = col.str().lower()
+    col = col.str().replace("\\+?(inf|infinity)$", "Inf", regex=True)
+    col = col.str().replace("-(inf|infinity)$", "-Inf", regex=True)
```

Comment:
You may be better off using two separate calls instead of a regular expression here. Regular expressions are pretty expensive. May want to run a quick benchmark to confirm.

Reply:
What's the "two separate calls" you referred to here?

Reply:
Offline benchmark result: https://github.com/isVoid/cudf-benchmarks/blob/master/str_repl_benchmark.ipynb